libvast/src/system/node.cpp

```diff
@@ -362,21 +363,50 @@ node_state::spawn_command(const invocation& inv,
     spawn_inv.options["import"] = import_opt;
     caf::put(spawn_inv.options, "vast.import", import_opt);
   }
-  // Spawn our new VAST component.
-  spawn_arguments args{spawn_inv, this_node->state.dir, label};
-  auto component = spawn_component(spawn_inv, args);
-  if (!component) {
-    if (component.error())
-      VAST_WARNING(__func__,
-                   "failed to spawn component:", render(component.error()));
-    rp.deliver(component.error());
-    return caf::make_message(component.error());
+  auto spawn_actually = [=](const invocation& spawn_inv) mutable {
+    // Spawn our new VAST component.
+    spawn_arguments args{spawn_inv, this_node->state.dir, label};
+    auto component = spawn_component(spawn_inv, args);
+    if (!component) {
+      if (component.error())
+        VAST_WARNING(
+          __func__, "failed to spawn component:", render(component.error()));
+      rp.deliver(component.error());
+      return;
+    }
+    this_node->monitor(*component);
+    auto okay = this_node->state.registry.add(*component, std::move(comp_type),
+                                              std::move(label));
+    VAST_ASSERT(okay);
+    rp.deliver(*component);
+    return;
+  };
+  auto handle_taxonomies = [=](taxonomies_ptr t) mutable {
```

Comment:
I think you have a race here. The pointer is effectively internal state in the type registry. What prevents the registry to mutate it while it's being looked here?

Reply:
This is does not apply anymore after the latest API change.